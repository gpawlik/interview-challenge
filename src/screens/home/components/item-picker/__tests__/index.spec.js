import * as React from "react";
import renderer from "react-test-renderer";

import { ItemPicker } from "../";

jest.mock("react-redux", () => ({
  useDispatch: () => {},
  useSelector: (item) => item(),
}));

jest.mock("../../../../../components/menu-item", () => ({
  MenuItem: "MenuItem",
}));

jest.mock("../../../../../state/menu/selectors", () => ({
  getAvailableItems: () => {
    return [
      { id: 1, name: "Text A", dietaries: [] },
      { id: 2, name: "Text B", dietaries: [] },
    ];
  },
}));

jest.mock("../styles", () => ({
  Container: "Container",
}));

const initialProps = {};

const mockComponent = (props) => {
  const finalProps = {
    ...initialProps,
    ...props,
  };

  return renderer.create(<ItemPicker {...finalProps} />);
};

describe("ItemPicker", () => {
  it("renders correctly default version", () => {
    const tree = mockComponent().toJSON();

    expect(tree).toMatchSnapshot();
  });
});
