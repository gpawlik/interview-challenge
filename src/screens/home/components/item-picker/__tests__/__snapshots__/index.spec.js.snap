// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemPicker renders correctly default version 1`] = `
<Container>
  <MenuItem
    dietaries={Array []}
    name="Text A"
    onPress={[Function]}
  />
  <MenuItem
    dietaries={Array []}
    name="Text B"
    onPress={[Function]}
  />
</Container>
`;
